src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use pyo3::ffi::{
+    PyContextVar_Get, PyEval_SetProfile, PyFrameObject, PyUnicode_AsUTF8, Py_tracefunc,
+    CO_ASYNC_GENERATOR, CO_COROUTINE, CO_ITERABLE_COROUTINE,
+};
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::{AsPyPointer, PyAny, Python};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::ffi::CStr;
+use std::os::raw::c_int;
+use std::time::Instant;
+
+/// Enum of possible Python's Trace/Profiling events
+#[allow(dead_code)]
+#[repr(i32)]
+enum TraceEvent {
+    Call,
+    Exception,
+    Line,
+    Return,
+    CCall,
+    CException,
+    CReturn,
+    Opcode,
+    __NonExhaustive,
+}
+
+impl TryFrom<c_int> for TraceEvent {
+    type Error = &'static str;
+    /// Cast i32 event (raw from Python) to Rust enum.
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        if value > 7 || value < 0 {
+            return Err("Not valid enum value");
+        }
+        Ok(unsafe { std::mem::transmute(value) })
+    }
+}
+
+/// Struct representing frame data
+/// Identifier is frame pointer casted to usize.
+struct FrameData {
+    func_name: String,
+    file_name: String,
+    identifier: usize,
+    yielded_coroutine: bool,
```

Comment:
consider `is_yielded_coroutine` or `did_yield_coroutine`, English is ambiguous otherwise :\